android-timetracker/src/com/markuspage/android/atimetracker/TaskNotificationThread.java

```diff
@@ -0,0 +1,254 @@
+        // to extend.
+
+        StringBuilder titleBuilder = new StringBuilder();
+        StringBuilder contentBuilder = new StringBuilder();
+
+        active = this.taskAdapter.findCurrentlyActive();
+
+        Log.d(TAG, "Running thread...");
+        debugPreferences();
+
+        // If tasks are running then update the notification
+        if (active.hasNext() && (notificationMode != NOTIFICATION_NEVER)){
+            Log.d(TAG, "Active Tasks:");
+
+            while (active.hasNext()) {
+                ++nTasks;
+                t = active.next();
+                Log.d(TAG, "\t- " + t.getTaskName());
+                contentBuilder.append("\t- ")
+                    .append(t.getTaskName())
+                    .append("\n");
+            }
+
+            // Construct the content and title
+            titleBuilder.append(String.valueOf(nTasks)).append(
+                    " task");
+            if (nTasks > 1)
+                titleBuilder.append("s");
+            titleBuilder.append(" running:");
+            //contentBuilder.insert(0, titleBuilder);
+            content = contentBuilder.toString();
+            title = titleBuilder.toString();
+
+            // Set the content and title
+            this.notificationBuilder
+                .setContentTitle(title)
+                .setContentText(content);
+
+            // TODO: Watch this, because sometimes we get a
+            // StackOverflowError.
+            this.notificationManager.notify(notificationId,
+                    this.notificationBuilder.build());
+        } else if (notificationMode == NOTIFICATION_ALWAYS){
+            titleBuilder.append("No tasks running");
+            contentBuilder.append("Click here to start a task.");
+            this.notificationBuilder
+                .setContentTitle(titleBuilder.toString())
+                .setContentText(contentBuilder.toString());
+            this.notificationManager.notify(notificationId,
+                    this.notificationBuilder.build());
+        } else {
+            this.notificationManager.cancel(notificationId);
+            Log.d(TAG, "No active tasks. Not setting notification.");
+        }
+    }
+
+    private void debugPreferences() {
+        switch(notificationMode){
+            case NOTIFICATION_ALWAYS:
+                Log.d(TAG, "\tNotification Mode: ALWAYS");
+                break;
+            case NOTIFICATION_ACTIVE:
+                Log.d(TAG, "\tNotification Mode: ACTIVE");
+                break;
+            case NOTIFICATION_NEVER:
+                Log.d(TAG, "\tNotification Mode: NEVER");
+                break;
+        }
+    }
+
+    @Override
+    public void run(){
+        while (true){
+            setNotification();
+
+            Log.d(TAG, "Running " + this);
+
+            // Finally, go to sleep and call run() again.
+            try {
+                Log.d(TAG, "Sleeping for " + SLEEP_TIME);
+                Thread.sleep(TaskNotificationThread.SLEEP_TIME);
```

Comment:
Hi, Marckus. As per the official android docs (https://developer.android.com/guide/topics/ui/notifiers/notifications.html) this is the way to start a notification. I set the time for 1 second because that's usually the interval for a clock but we can make it longer for more of a "fuzzy" time if you'd like.